### PR TITLE
Update core tests: fixes for CRLF in multiline tests, executable-not-found issue

### PR DESCRIPTION
**Pull request by cxw42**
See editorconfig/editorconfig-core-test#28

CC @xuhdev

### CODE REVIEW

**Comment by cxw42** [comment]
@xuhdev Have you seen this failure before? --- https://travis-ci.org/editorconfig/editorconfig-core-c/jobs/600209206 .  pcre2 static-link failure on Xenial with GCC.  I checked, and [libpcre2-dev on xenial](https://packages.ubuntu.com/xenial/amd64/libpcre2-dev/filelist) does have a static library.

**Comment by xuhdev** [comment]
Awesome! This is really good work! :fireworks: 

No I haven't seen that error before. I'm enabling verbose output in #64 so we can see what's going on.

**Comment by xuhdev** [comment]
But path_separator_backslash_in_cmd_line fails now on Windows, which did not fail before...

I'll try to fix the static build issue (which shouldn't be relevant here). Something may have changed on Travis...

**Comment by cxw42** [comment]
@xuhdev Thanks for looking into the static-build issue!

I am continuing to investigate the backslashes.  The problem with the Windows builds seems to be twofold:

- CTest is checking its input strings for backslash escapes, and warning if they aren't valid, so we need `` \\ ``.
- But CTest is also passing the strings through unchanged to the shell, so we need `` \ ``!  

I am checking whether I can move the `$BackSlashes` variable where CMake won't process it, so CTest can do so.

**Comment by cxw42** [comment]
@xuhdev I am logging off for now and plan to work on this again tomorrow. o/

**Comment by xuhdev** [comment]
Thanks! #69 should have resolved the static build issue -- feel free to rebase

**Comment by cxw42** [comment]
@xuhdev Yep, Travis build passed :) .  Thanks for the quick fix!

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,4 @@
 [submodule "tests"]
 	path = tests
-	url = git://github.com/editorconfig/editorconfig-core-test.git
+	url = https://github.com/cxw42/editorconfig-core-test.git
```
Comment (xuhdev) [inline .gitmodules]: Shall we update the submodule to the latest test repo and get this merged?

Comment (cxw42) [reply .gitmodules]: All set!

**Comment by xuhdev** [comment]
:fireworks: :fireworks: :fireworks:

**Pull request merged**